open_spiel/python/algorithms/tabular_maqlearner.py

```diff
@@ -0,0 +1,209 @@
+# Copyright 2019 DeepMind Technologies Limited
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Tabular Multiagent Q-learning agent."""
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import abc
+import collections
+import itertools
+import numpy as np
+
+from open_spiel.python import rl_agent
+from open_spiel.python import rl_tools
+from open_spiel.python.algorithms.jpsro import _mgce, _mgcce
+from open_spiel.python.examples.matrix_nash import lemke_howson_solve
+
+def valuedict():
+  return collections.defaultdict(float)
+
+
+class JointActionSolver:
+  @abc.abstractmethod
+  def __call__(self, q_tables, action_dims):
+    '''
+      Args:
+        q_tables: a list of Q tables for each agent
+        action_dims: a list of ints recording the size of action spaces for each agent
+      Returns:
+        res_mixtures: a list of mixed strategies for each agent
+        res_values: a list of expected utilities for each agent
+    '''
+    
+
+class TwoPlayerNashSolver(JointActionSolver):
+  '''
```

Comment:
Style: can you use the delimiter and header style which consistent delimiter with the rest of the code base? For one-line function headers: 

"""Use three double quotations and comment on the same line."""
(followed by one blank line)

Similarly throughout all files.